src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -172,9 +172,9 @@ void symbol_factoryt::gen_nondet_array_init(
   auto const &array_type = to_array_type(expr.type());
   const auto &size = array_type.size();
   PRECONDITION(size.id() == ID_constant);
-  auto const array_size = numeric_cast_v<size_t>(to_constant_expr(size));
+  auto const array_size = numeric_cast_v<mp_integer>(to_constant_expr(size));
   DATA_INVARIANT(array_size >= 0, "Arrays must have non-negative size");
-  for(size_t index = 0; index < array_size; ++index)
+  for(mp_integer index = 0; index < array_size; ++index)
```

Comment:
While this makes sense in spirit, does this actually make a difference in practice (other than slightly worse performance)? We wouldn't be able to allocate memory for an object larger than max(size_t) anyway.